### PR TITLE
Flag isSelectedItemStraightened is set to false at the end of straightening

### LINKED ISSUE #2: FanLayoutManager::straightenSelectedItem is not reseting isSelectedItemStraightened at the end of animation
Hello!

As I can see, the member variable `isSelectedItemStraightened` is used to indicate if the selected item is being straightened at the moment. It is set to `true` at the beginning of animation and in my opinion shall be set to false at the end.

### PR DESCRIPTION
**Pull request by pchmielowski**
Hello! Here is my proposal to resolve #2

### CODE REVIEW

File: library/src/main/java/com/cleveroad/fanlayoutmanager/AnimationHelper.java [hunk 1]
```diff
@@ -57,8 +57,11 @@ void shiftSideViews(@NonNull final Collection<ViewAnimationInfo> views,
...
-     * @param view     view to rotate
-     * @param listener animator listener to check start or end animation
+     * @param view            view to rotate
+     * @param customListener  custom animator listener to check start or end animation
+     * @param managerListener animator listener to notify layout manager of the animation state
      */
-    void straightenView(View view, @Nullable Animator.AnimatorListener listener);
+    void straightenView(View view,
+                        @Nullable Animator.AnimatorListener customListener,
+                        @Nullable Animator.AnimatorListener managerListener);
```
Comment (Iojjj) [inline library/src/main/java/com/cleveroad/fanlayoutmanager/AnimationHelper.java:66]: If you need to use multiple listeners you can pass your own implementation of composite listener here (a listener that holds references to the list of another listeners and calls them once some event is triggered). See the comment with an example below.

File: library/src/main/java/com/cleveroad/fanlayoutmanager/FanLayoutManager.java [hunk 2]
```diff
@@ -874,7 +874,28 @@ public void straightenSelectedItem(Animator.AnimatorListener listener) {
         if (viewToRotate != null) {
 
             // start straight animation
-            animationHelper.straightenView(viewToRotate, listener);
+            animationHelper.straightenView(viewToRotate, listener, new
```
Comment (Iojjj) [inline library/src/main/java/com/cleveroad/fanlayoutmanager/FanLayoutManager.java:877]: You don't need to pass a new instance just for changing the `isSelectedItemStraightened` flag. Just wrap an existing `listener` with a new one. An example of better approach:

``` JAVA
animationHelper.straightenView(viewToRotate, new AnimatorListenerAdapter() { 
    @Override
    public void onAnimationStart(Animator animation) {
        listener.onAnimationStart(animation);
    }

    @Override
    public void onAnimationEnd(Animator animation) {
        isSelectedItemStraightened = false;
        listener.onAnimationEnd(animation);
    }

    @Override
    public void onAnimationCancel(Animator animation) {
        listener.onAnimationCancel(animation);
    }

    @Override
    public void onAnimationRepeat(Animator animation) {
        listener.onAnimationRepeat(animation);
    }
});
```

**Comment by Iojjj** [comment]
We can't accept this pull request. Please make a pull request to `develop` branch only.

**Comment by yarovoiag** [comment]
Was implemented in version 1.0.2

**Pull request closed without merging**